### ISSUE TITLE
"do if" transpiles to "doif"

### ISSUE THREAD

**Issue by dgreensp** [babel/babel#1290]
With `compact: true` specified, this:

```
do if (x) y(); while (false);
```

transpiles to this:

```
doif(x)y();while(false);
```

which is not valid JavaScript (`doif`).  This causes Babel to crash on certain generated JS files.

**Comment by sebmck** [comment]
That is some extremely funky code but the attached patch should fix this.

**Comment by dgreensp** [comment]
Heh. :)  My example was a little too funky in retrospect.  It could have been, for example:

```
do console.log(iter.current()); while (iter.advance());
```

Emscripten seems to generate code like:

```
do if(k<<24>>24) {
  m=0;
  while(1) {
    n=m;
    m=m+1|0;
...
```

Anyway, thanks!

**Comment by sebmck** [comment]
Fixed as of 5.1.11, thanks!
